### PR TITLE
[FLINK-2248][client]Add flag to disable sysout logging from cli

### PR DESCRIPTION
**Pull request by sachingoel0101**
Enables disabling of sysout messages on cli via a flag `q`.

### CODE REVIEW

File: flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java [hunk 1]
```diff
@@ -51,6 +51,9 @@
 			"The parallelism with which to run the program. Optional flag to override the default value " +
 					"specified in the configuration.");
 
+	static final Option LOGGING_OPTION = new Option("q","stdoutLogging", false, "Whether stdout " +
```
Comment (mxm) [inline flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java]: Space missing after `"q",`.

**Comment by mxm** [comment]
Thanks for the pull request. Could you add a test case to check whether the output is suppressed when `-q` is provided? Also, it would be great if you could update the documentation with a hint to the new flag.

**Comment by sachingoel0101** [comment]
@mxm , I've added unit tests to verify disabled logging.

File: flink-clients/src/test/java/org/apache/flink/client/CliFrontendLoggingTest.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	private static LocalFlinkMiniCluster cluster;
+	private static Configuration config;
+	private static String hostPort;
+	private ByteArrayOutputStream stream = new ByteArrayOutputStream();
+	private CliFrontend cli;
+
+	@Before
+	public void setUp() throws Exception {
+		stream.reset();
+		System.setOut(new PrintStream(stream));
```
Comment (mxm) [inline flink-clients/src/test/java/org/apache/flink/client/CliFrontendLoggingTest.java]: You are overwriting stdout permanently here. You could save `System.out` to a variable here and set it again in the `@After` method.

**Comment by mxm** [comment]
Thanks for adding the tests! This looks good to merge. Could you update https://ci.apache.org/projects/flink/flink-docs-master/apis/cli.html ?

**Comment by sachingoel0101** [comment]
@mxm , updated.

**Comment by mxm** [comment]
Thanks. One more thing. Could you also update https://ci.apache.org/projects/flink/flink-docs-release-0.9/setup/config.html ? You introduced a new config entry.

**Comment by sachingoel0101** [comment]
Done.

**Comment by mxm** [comment]
Thank you! Merging.

**Comment by mxm** [comment]
Actually, I'm not sure about the config entry. We already have a lot of config entries and I don't really see the purpose of making print output of the client a system wide setting. What was your rational behind introducing the config parameter?

**Comment by sachingoel0101** [comment]
There was no specific need for it. However, since the CliFrontend only passes the Client the configuration, I decided to include it with that.
Further, I think while writing programs using flink [flink-example-esqe] and running them in IDE, it would  be nice to have a config entry to turn off the log. It gets distracting sometimes.

[Although this PR doesn't quite achieve that. That would require some changes in the Local Executor].

**Comment by sachingoel0101** [comment]
yarnfifo case failure. 
nothing to do with the changes made in this PR.

File: flink-clients/src/test/java/org/apache/flink/client/CliFrontendLoggingTest.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+			e.printStackTrace();
+			fail("Setup of test actor system failed.");
+		}
+
+		cli = new CliFrontend(getConfigDir());
+	}
+
+	@After
+	public void shutDownActorSystem() {
+		System.setOut(new PrintStream(output));
```
Comment (mxm) [inline flink-clients/src/test/java/org/apache/flink/client/CliFrontendLoggingTest.java]: I think it should be `System.setOut(output)` here.

Comment (sachingoel0101) [inline flink-clients/src/test/java/org/apache/flink/client/CliFrontendLoggingTest.java]: System.setOut only takes a PrintStream as argument.

**Comment by mxm** [comment]
> There was no specific need for it. However, since the CliFrontend only passes the Client the configuration, I decided to include it with that.

There is a method on the Client to set the printing.

>Further, I think while writing programs using flink [flink-example-esqe] and running them in IDE, it would be nice to have a config entry to turn off the log. It gets distracting sometimes.

If you want to achieve that, you can adjust the log4j.properties file and set logging to OFF.

I would be for excluding the config entry because I don't think people will use it.

File: flink-clients/src/test/java/org/apache/flink/client/CliFrontendLoggingTest.java (hunk 3 above)
Comment (mxm) [inline flink-clients/src/test/java/org/apache/flink/client/CliFrontendLoggingTest.java]: Yes, and System.out returns a PrintStream :)

Comment (sachingoel0101) [inline flink-clients/src/test/java/org/apache/flink/client/CliFrontendLoggingTest.java]: Ah. I just assumed it would be some kind of OutputStream.

**Comment by sachingoel0101** [comment]
Yeah. I think we should remove it then. Too many entries tend to confuse people.

**Comment by sachingoel0101** [comment]
I've updated the code to remove any changes in Configuration.

**Comment by mxm** [comment]
Thank you for your patience :) I will merge this once the tests have passed.

**Comment by sachingoel0101** [comment]
Sure. No problem. :)

**Comment by sachingoel0101** [comment]
Travis build passes. You can merge it @mxm

**Pull request closed without merging**